web/src/hashchange.js

```diff
@@ -257,12 +264,12 @@ function do_hashchange_overlay(old_hash) {
         if (base === "streams") {
             // e.g. #streams/29/social/subscribers
             const right_side_tab = hash_parser.get_current_nth_hash_section(3);
-            stream_settings_ui.change_state(section, right_side_tab);
+            stream_settings_ui.change_state(section, undefined, right_side_tab);
             return;
         }
 
         if (base === "groups") {
-            user_group_edit.change_state(section);
+            user_group_edit.change_state(section, undefined);
```

Comment:
Possible bug? But from what i've tested, switching streams/groups using buttons in stream or group settings overlay doesn't trigger this. However, if navigated using links (by manually editing browser url or links in stream description), selected tab may be reset.

Reply:
For this, we can either get current toggle value and pass that OR add a possible value for `left_side_tab` like "preserve" that turns it into a no-op. The former looks cleaner, we will need to export `group_list_toggle` from `user_group_edit.js` but thats fine i think?

Reply:
I don't get what is the bug here? Can you elaborate.

Reply:
We pass `left_side_tab` as undefined in `change_state` to let it select the tab based on user's membership. So `change_state` might switch to other tab if group isn't in the currently selected tab.

For example, if I click on some link in description of a stream I am subscribed to AND that link points to an unsubscribed stream, then selected tab will change to "All streams" and list will be redrawn.

Although, I am not sure if group descriptions can contain links so might not be a problem here. But above example with streams I have tested.

Reply:
To preserve current behaviour and simplify manual testing; I've made this modification for now.

```diff
- stream_settings_ui.change_state(section, right_side_tab);
+ stream_settings_ui.change_state(section, "preserve", right_side_tab);
```

Reply:
> For example, if I click on some link in description of a stream I am subscribed to AND that link points to an unsubscribed stream, then selected tab will change to "All streams" and list will be redrawn.

Yeah, so this works fine as expected I guess. We want to show the tab based on whether the user is subscribed to the stream opened in right panel or not.

Reply:
Modified this to show appropriate tab based on membership/subscription to the group/stream being opened. Please take a look when you can @sahil839. Thanks!